src/integer_mod_q/mat_polynomial_ring_zq/default.rs

```diff
@@ -103,3 +134,106 @@ mod test_new {
         let _ = MatPolynomialRingZq::new(2, 2, &modulus);
     }
 }
+
+#[cfg(test)]
+mod test_identity {
+    use crate::{
+        integer::PolyOverZ,
+        integer_mod_q::{MatPolynomialRingZq, ModulusPolynomialRingZq},
+        traits::GetEntry,
+    };
+    use std::str::FromStr;
+
+    /// Tests if an identity matrix is set from a zero matrix.
+    #[test]
+    fn identity() {
+        let modulus = ModulusPolynomialRingZq::from_str("3  1 0 1 mod 17").unwrap();
+        let matrix = MatPolynomialRingZq::identity(10, 10, &modulus);
+
+        for i in 0..10 {
+            for j in 0..10 {
+                if i != j {
+                    assert_eq!(PolyOverZ::default(), matrix.get_entry(i, j).unwrap());
+                } else {
+                    assert_eq!(
+                        PolyOverZ::from_str("1  1").unwrap(),
+                        matrix.get_entry(i, j).unwrap()
+                    )
+                }
+            }
+        }
+    }
+
+    /// Tests if function works for a non-square matrix.
+    #[test]
+    fn non_square_works() {
+        let modulus = ModulusPolynomialRingZq::from_str("3  1 0 1 mod 17").unwrap();
+        let matrix = MatPolynomialRingZq::identity(10, 7, &modulus);
+
+        for i in 0..10 {
+            for j in 0..7 {
+                if i != j {
+                    assert_eq!(PolyOverZ::default(), matrix.get_entry(i, j).unwrap());
+                } else {
+                    assert_eq!(
+                        PolyOverZ::from_str("1  1").unwrap(),
+                        matrix.get_entry(i, j).unwrap()
+                    )
+                }
+            }
+        }
+
+        let matrix = MatPolynomialRingZq::identity(7, 10, &modulus);
+
+        for i in 0..7 {
+            for j in 0..10 {
+                if i != j {
+                    assert_eq!(PolyOverZ::default(), matrix.get_entry(i, j).unwrap());
+                } else {
+                    assert_eq!(
+                        PolyOverZ::from_str("1  1").unwrap(),
+                        matrix.get_entry(i, j).unwrap()
+                    )
+                }
+            }
+        }
+    }
+
+    /// Tests if an identity matrix can be created using a large modulus.
+    #[test]
+    fn modulus_large() {
+        let modulus =
+            ModulusPolynomialRingZq::from_str(&format!("3  1 {} 1 mod {}", i64::MAX, u64::MAX))
```

Comment:
```suggestion
            ModulusPolynomialRingZq::from_str(&format!("4  1 0 {} 1 mod {}", i64::MAX, u64::MAX))
```
Just to add a test for another polynomial dimension